wgpu-hal/src/dx12/adapter.rs

```diff
@@ -381,17 +381,18 @@ impl crate::Adapter<super::Api> for super::Adapter {
         // the features that use SRV/UAVs using the no-depth format.
         let mut data_no_depth = d3d12::D3D12_FEATURE_DATA_FORMAT_SUPPORT {
             Format: no_depth_format,
-            Support1: mem::zeroed(),
-            Support2: mem::zeroed(),
+            Support1: d3d12::D3D12_FORMAT_SUPPORT1_NONE,
+            Support2: d3d12::D3D12_FORMAT_SUPPORT2_NONE,
         };
         if raw_format != no_depth_format {
             // Only-recheck if we're using a different format
             assert_eq!(
                 winerror::S_OK,
                 self.device.CheckFeatureSupport(
                     d3d12::D3D12_FEATURE_FORMAT_SUPPORT,
-                    &mut data_no_depth as *mut _ as *mut _,
-                    mem::size_of::<d3d12::D3D12_FEATURE_DATA_FORMAT_SUPPORT>() as _,
+                    ptr::addr_of_mut!(data_no_depth).cast::<c_void>(),
+                    DWORD::try_from(mem::size_of::<d3d12::D3D12_FEATURE_DATA_FORMAT_SUPPORT>())
```

Comment:
My only hesitation is the verbosity of this

Reply:
Coming back to this with a fresh brain:

1. RE: pointer slinging:
	1. We can definitely omit the explicit `::<c_void>`. I've done so with 7c46b4b824e69b06df1d60b0b48c563b14251986.
	1. We could fully qualify the `addr_of_mut` symbol, which would make this shorter and, IMO, easier to follow. In general, I think a _lot_ of this code would be smaller with no loss of clarity (and therefore more friendly to the mental cache) without path qualifications for most symbols.
1. I've seen folks like @retep998 make utility crates for Windows specifically to convert `usize`s to `DWORD`s. Maybe adding something like an extension trait for this in a separate commit would be good?

Thoughts?

Reply:
Honestly now that I look at this, this seems fine.